src/java.base/share/classes/java/io/StreamTokenizer.java

```diff
@@ -791,43 +778,30 @@ public int lineno() {
      * @see     java.io.StreamTokenizer#ttype
      */
     public String toString() {
-        String ret;
-        switch (ttype) {
-          case TT_EOF:
-            ret = "EOF";
-            break;
-          case TT_EOL:
-            ret = "EOL";
-            break;
-          case TT_WORD:
-            ret = sval;
-            break;
-          case TT_NUMBER:
-            ret = "n=" + nval;
-            break;
-          case TT_NOTHING:
-            ret = "NOTHING";
-            break;
-          default: {
+
+        String ret = switch (ttype) {
+            case TT_EOF     -> ret = "EOF";
+            case TT_EOL     -> ret = "EOL";
+            case TT_WORD    -> ret = sval;
+            case TT_NUMBER  -> ret = "n=" + nval;
+            case TT_NOTHING -> ret = "NOTHING";
```

Comment:
This is not correct, the assignments to ret in these case arms is redundant.

Reply:
Comment addressed in e503ed2